app/controllers/homeworks_controller.rb

```diff
@@ -1,6 +1,6 @@
 class HomeworksController < ApplicationController
   before_action :set_homework, only: [:show, :edit, :update, :destroy]
-  before_action :set_lecture, only: [:new, :create]
+  before_action :set_lecture, only: [:new]
   respond_to :html, :json
```

Comment:
I think this is what is causing the site error - this statement should be inside an action, but really probably doesn't need to be here at all.

```suggestion
```